packages/upset/src/components/AltTextSidebar.tsx

```diff
@@ -24,37 +28,48 @@ type Props = {
 const initialDrawerWidth = 450;
 
 export const AltTextSidebar: FC<Props> = ({ open, close, generateAltText }) => {
-  const [altText, setAltText] = useState<string>('');
-  const [verbosity, setVerbosity] = useState<string>('low');
-  const [explain, setExplain] = useState<string>('full');
+  const { provenance, actions } = useContext(ProvenanceContext);
+  const { verbosity, explain } = useRecoilValue(altTextSelector);
+  const setGenerating = useSetRecoilState(generatingAtom);
 
-  const { provenance } = useContext(ProvenanceContext);
+  const [textDescription, setTextDescription] = useState('');
 
-  function genAltText(): void {
-    async function generate(): Promise<void> {
-      const resp = await generateAltText(verbosity, explain);
-      setAltText(resp);
-    }
-
-    generate();
+  async function genAltText(verb: string, exp: string): Promise<any> {
+    return generateAltText(verb, exp);
   }
 
-  useEffect(() => {
-    genAltText();
-  }, [verbosity, explain]);
-
   provenance.currentChange(() => {
-    genAltText();
+    // if the current node is not setVerbosity, setExplain, or setTextDescription (not in altText prov object)
+    if (provenance.current.event !== 'set-verbosity' && provenance.current.event !== 'set-explain') {
+      setGenerating(true);
+      genAltText(verbosity, explain).then((resp) => {
+        setTextDescription(resp);
+      });
+    }
+    setGenerating(false);
   });
 
   const handleVerbosityChange = (e: EventTarget & HTMLInputElement): void => {
-    setVerbosity(e.value);
+    setGenerating(true);
+    genAltText(e.value, explain).then((resp) => {
+      setTextDescription(resp);
+      actions.setVerbosity(e.value);
+      setGenerating(false);
+    });
   };
 
   const handleExplainChange = (e: EventTarget & HTMLInputElement): void => {
-    setExplain(e.value);
+    setGenerating(true);
+    genAltText(verbosity, e.value).then((resp) => {
+      setTextDescription(resp);
+      actions.setExplain(e.value);
+      setGenerating(false);
+    });
   };
```

Comment:
Why not just make these variables part of the trrack store? Then it can all be handled by that first block

Reply:
This was because the provenance change was not recognizing the changed value, rather it was using the previous version and then updating the value. This is fixed with the recent `useEffect` changes.